alti/subscribers.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+
+from pyramid.events import subscriber, NewRequest
+from pyramid.httpexceptions import HTTPRequestEntityTooLarge
+
+
+@subscriber(NewRequest)
+def validate_body_length(event):
+    max_clength = 160000
+    content_length = event.request.headers.get('Content-Length')
+    content_length = content_length or len(event.request.params.get('geom', ''))
```

Comment:
Does this cover POST requests?

Does this check content-lenght before the requests is parsed (POST body)?

I think we could say that for profile requests, we ignore requests that don't have a content-length header set at all when POST

Reply:
We can use both GET and POST, here we validate both geom and/or length of geom parameter.
> I think we could say that for profile requests, we ignore requests that don't have a content-length header set at all when POST

I can add this check, clients usually implement it.

Reply:
I think there are limits on the GET params length anyway, either on apache, varnish or other proxy levels. Heck, even IE has a limit for total URL lenght.

My worry is that we need to parse the content to determine its length - and that the parsing of the content is slow. But I have not tested this.

Reply:
IE9 has a maxium of 2048.
It really varies according to the browser here is a complete list.
https://boutell.com/newfaq/misc/urllength.html

Reply:
Note that this happend before entering the view, so no json parsing here, it's before.
I can't do it at Apache level, which was my first attempt.